codegenerator/cli/templates/dynamic/codegen/src/Types.res.hbs

```diff
@@ -383,55 +459,126 @@ module type Event = {
   let name: string
   let contractName: string
-  let chains: array<ChainMap.Chain.t>
 
   type eventArgs
   let eventArgsSchema: S.schema<eventArgs>
   let convertHyperSyncEventArgs: HyperSyncClient.Decoder.decodedEvent => eventArgs
   let decodeHyperFuelData: string => eventArgs
   let handlerRegister: HandlerTypes.Register.t<eventArgs>
+
+  type eventFilter
+  let getTopicSelection: SingleOrMultiple.t<eventFilter> => array<LogSelection.topicSelection>
 }
 module type InternalEvent = Event with type eventArgs = internalEventArgs
 
 external eventToInternal: eventLog<'a> => eventLog<internalEventArgs> = "%identity"
 external eventModToInternal: module(Event with type eventArgs = 'a) => module(InternalEvent) = "%identity"
 external eventModWithoutArgTypeToInternal: module(Event) => module(InternalEvent) = "%identity"
 
+let makeEventOptions = (
+  type eventFilter,
+  eventConfig: option<HandlerTypes.eventConfig<eventFilter>>,
+  eventMod: module(Event with type eventFilter = eventFilter),
+) => {
+  let module(Event) = eventMod
+  open Belt
+  eventConfig->Option.map(({?wildcard, ?eventFilters}) =>
+    HandlerTypes.EventOptions.make(
+      ~isWildcard=wildcard->Option.getWithDefault(false),
+      ~topicSelections=eventFilters->Option.mapWithDefault(
+        [
+          LogSelection.makeTopicSelection(
+            ~topic0=[Event.sighash->EvmTypes.Hex.fromStringUnsafe],
+          )->Utils.unwrapResultExn,
+        ],
+        v => v->Event.getTopicSelection,
+      ),
+    )
+  )
+}
+
+let makeGetEventOptions = (
+  type eventFilter eventArgs,
+  eventMod: module(Event with type eventFilter = eventFilter and type eventArgs = eventArgs),
+) => {
+  open Belt
+  let module(Event) = eventMod
+  (loaderHandler: HandlerTypes.loaderHandler<Event.eventArgs, 'loaderReturn, Event.eventFilter>) =>
+    switch loaderHandler {
+    | {wildcard: ?None, eventFilters: ?None} => None
+    | {?wildcard, ?eventFilters} =>
+      let topicSelections =
+        eventFilters->Option.mapWithDefault(
+          [
+            LogSelection.makeTopicSelection(
+              ~topic0=[Event.sighash->EvmTypes.Hex.fromStringUnsafe],
+            )->Utils.unwrapResultExn,
+          ],
+          v => v->Event.getTopicSelection,
+        )
+      HandlerTypes.EventOptions.make(
+        ~isWildcard=wildcard->Option.getWithDefault(false),
+        ~topicSelections,
+      )->Some
+    }
+}
+
+@genType.import(("./bindings/OpaqueTypes.ts", "HandlerWithOptions"))
+type fnWithEventConfig<'fn, 'eventConfig> = ('fn, ~eventConfig: 'eventConfig=?) => unit
+
+@genType
+type handlerWithOptions<'eventArgs, 'loaderReturn, 'eventFilter> = fnWithEventConfig<
+  HandlerTypes.handler<'eventArgs, 'loaderReturn>,
+  HandlerTypes.eventConfig<'eventFilter>,
+>
+
+@genType
+type contractRegisterWithOptions<'eventArgs, 'eventFilter> = fnWithEventConfig<
+  HandlerTypes.contractRegister<'eventArgs>,
+  HandlerTypes.eventConfig<'eventFilter>,
+>
```

Comment:
Here is the trick I used to allow for an optional second argument in typescript. Only works for "handler" and "contractRegister"